api/v1/errors/SqlError.js

```diff
@@ -0,0 +1,28 @@
+function checkError(err) {
```

Comment:
Ultimately I'm fine with you keeping a SQLError somewhere as long as the client doesn't get to see it purposefully (hence why I said elsewhere that SQLError should be a direct descendant of APIError). It should be caught and re-thrown as an InvalidParameter error with a well-formed description (and source if applicable).

The client should never see the SQL error generated.